src/ds5/ds5-options.cpp

```diff
@@ -655,24 +655,24 @@ namespace librealsense
             return _uvc_option->is_enabled();
     }
 
-    auto_exposure_limit_option::auto_exposure_limit_option(hw_monitor& hwm, sensor_base* ep, option_range range)
-        : option_base(range), _hwm(hwm), _sensor(ep)
+    auto_exposure_limit_option::auto_exposure_limit_option(hw_monitor& hwm, sensor_base* ep, option_range range, limits_option* exposure_limit_enable)
+        : option_base(range), _hwm(hwm), _sensor(ep), _exposure_limit_enable(exposure_limit_enable)
     {
         _range = [range]()
         {
             return range;
         };
-        _cached_limit = range.max;
-        _enable_limit = 0; // setting limit is disabled by default
+        set(get_range().max);
     }
 
     void auto_exposure_limit_option::set(float value)
     {
-        if (!is_valid(value) && value != 0)
+        if (!is_valid(value))
             throw invalid_value_exception("set(enable_auto_exposure) failed! Invalid Auto-Exposure mode request " + std::to_string(value));
-        
-        if (_enable_limit != 0.f) // cache max limit only if setting limit is enabled
-            _cached_limit = value;
+
+        _exposure_limit_enable->set_cached_limit(value);
+        if (value != get_range().max && _exposure_limit_enable->query() == 0.f)
```

Comment:
I wanted to keep the toggle off at the beginning but I can remove the max-value condition.
The second part of the condition is relevant to prevent redundant commands to FW.